features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/ViewEndedMetricDispatcher.kt

```diff
@@ -0,0 +1,162 @@
+            buildMap {
+                put(KEY_VALUE, invState.initializationTime)
+                put(KEY_CONFIG, toAttributeValue(invState.config))
+                if (invState.initializationTime == null) {
+                    putNonNull(KEY_NO_VALUE_REASON, toAttributeValue(invState.noValueReason))
+                }
+            }
+        )
+    }
+
+    companion object {
+        const val DEFAULT_SAMPLE_RATE: Float = 0.75f
+
+        internal const val VIEW_ENDED_MESSAGE = "[Mobile Metric] RUM View Ended"
+
+        internal const val KEY_METRIC_TYPE = "metric_type"
+
+        private const val VALUE_METRIC_TYPE = "rum view ended"
+
+        internal const val KEY_RVE = "rve"
+        internal const val KEY_DURATION = "duration"
+        internal const val KEY_LOADING_TIME = "loading_time_value"
+
+        internal const val KEY_TIME_TO_NETWORK_SETTLED = "tns"
+        internal const val KEY_VALUE = "value"
+        internal const val KEY_CONFIG = "config"
+        private const val VALUE_TIME_BASED_DEFAULT = "time_based_default"
+        private const val VALUE_TIME_BASED_CUSTOM = "time_based_custom"
+
+        internal const val KEY_NO_VALUE_REASON = "no_value_reason"
+        private const val VALUE_NO_RESOURCES = "no_resources"
+        private const val VALUE_NO_INITIAL_RESOURCES = "no_initial_resources"
+        private const val VALUE_NOT_SETTLED_YET = "not_settled_yet"
+        private const val VALUE_UNKNOWN = "unknown"
+
+        internal const val KEY_INTERACTION_TO_NEXT_VIEW = "inv"
+        private const val VALUE_NO_ACTION = "no_action"
+        private const val VALUE_NO_ELIGIBLE_ACTION = "no_eligible_action"
+        private const val VALUE_NO_PREVIOUS_VIEW = "no_previous_view"
+
+        internal const val KEY_VIEW_TYPE = "view_type"
+        private const val VALUE_APPLICATION_LAUNCH = "application_launch"
+        private const val VALUE_BACKGROUND = "background"
+        private const val VALUE_CUSTOM = "custom"
+
+        private fun <K, V> MutableMap<K, V>.putNonNull(key: K, value: V?) {
+            if (value != null) put(key, value)
+        }
+
+        @JvmStatic
+        @VisibleForTesting
+        internal fun toAttributeValue(viewType: ViewMetricDispatcher.ViewType) = when (viewType) {
+            ViewMetricDispatcher.ViewType.CUSTOM -> VALUE_CUSTOM
+            ViewMetricDispatcher.ViewType.BACKGROUND -> VALUE_BACKGROUND
+            ViewMetricDispatcher.ViewType.APPLICATION -> VALUE_APPLICATION_LAUNCH
+        }
+
+        @JvmStatic
+        @VisibleForTesting
+        internal fun toAttributeValue(config: ViewInitializationMetricsConfig): String = when (config) {
+            ViewInitializationMetricsConfig.CUSTOM -> VALUE_CUSTOM
+            ViewInitializationMetricsConfig.TIME_BASED_DEFAULT -> VALUE_TIME_BASED_DEFAULT
+            ViewInitializationMetricsConfig.TIME_BASED_CUSTOM -> VALUE_TIME_BASED_CUSTOM
+        }
+
+        @JvmStatic
+        @VisibleForTesting
```

Comment:
why `JvmStatic` attribute is needed here, if they are supposed only be used from Kotlin?